api/defaults/defaults.go

```diff
@@ -34,6 +35,9 @@ const (
 	// DefaultIdleTimeout is a default idle connection timeout.
 	DefaultIdleTimeout = 30 * time.Second
 
+	// MinTLSVersion is the minimum TLS version the server will accept.
```

Comment:
The godoc here mentions only servers, but you're using this value when configuring clients.

In fact, I view this change as _weakening_ client side security. Go already defaults to minimum of 1.2 for TLS clients when `MinVersion` is not set. If at some point in the future, TLS 1.2 is no longer considered secure, Go may well change the default, and we would no longer be protected by the default behavior since we've explicitly opted in to TLS 1.2 everywhere.

Reply:
> but you're using this value when configuring clients

If so, that was unintentional.  I intended to reference this field for servers and for clients to reference `tls.VersionTLS12` directly.

> Go already defaults to minimum of 1.2 for TLS clients when MinVersion is not set

I did not realize this, I agree with you that we should remove this from client configurations.

Do you see anywhere I accidentally set `apidefaults.MinTLSVersion` in a client side configuration?  If not, I will just remove the `tls.` referenced changes.

Reply:
Ok, looking better, but now I wonder if the api/ module is the right place for this constant.

Our API module is designed to build clients, so why does it need a constant for how to configure servers?

Reply:
I put it here because it co-locates with `DefaultIOTimeout` and `DefaultIdleTimeout`.  These are also server configuration, but I believed they existed here to inform clients of this server configuration.  I think the TLS version configuration may make sense in the same way, particularly if / when we update to version 1.3.

Let me know your thoughts, I feel this is most consistent with the existing parameters.  But happy to move it.

Reply:
I'd like to see it moved. A constant in a module that is never used in said module feels like a code smell.